codec-http2/src/test/java/io/netty/handler/codec/http2/DataCompressionHttp2Test.java

```diff
@@ -148,9 +149,10 @@ public void gzipEncodingSingleEmptyMessage() throws Exception {
             FrameAdapter.getOrCreateStream(clientConnection, 3, false);
             runInChannel(clientChannel, new Http2Runnable() {
                 @Override
-                public void run() {
+                public void run() throws Http2Exception {
                     clientEncoder.writeHeaders(ctxClient(), 3, headers, 0, false, newPromiseClient());
                     clientEncoder.writeData(ctxClient(), 3, data.retain(), 0, true, newPromiseClient());
+                    clientConnection.remote().flowController().writePendingBytes();
```

Comment:
Maybe instead of this line (and the following line). Save a reference to the appropriate `Http2ConnectionHandler` and call flush on that (which should take care of all the underlying flushes)?